### PR TITLE
Thrusters, Sim, Diagnostics, Utils

### PR DESCRIPTION
**Pull request by DSsoto**
* Unifies thruster configuration for both simulated and real thrusters (removes a lot of outdated config files)
* Adds a utility to quickly conduct thruster spin tests
* Adds checking if the front cameras are connected to the usb bus (self check)
* Make the self check specifically list which thrusters are missing

### CODE REVIEW

**Comment by DSsoto** [comment]
Before anyone calls me out on it, I see that I've broken the `test_thruster_comm` test. I will fix before merge. Sweet, sweet CI !!

**Comment by kev-the-dev** [comment]
If you merge #222 would get rid of that little check mark for ya ;)

File: command/sub8_launch/nodes/self_check.py
Comment (mattlangford) [inline command/sub8_launch/nodes/self_check.py:155]: Do you have to have the first two commands here? Won't just calling lsusb -d work?

File: drivers/sub8_videoray_m5_thruster/nodes/thruster_spin_test.py
Comment (mattlangford) [inline drivers/sub8_videoray_m5_thruster/nodes/thruster_spin_test.py]: Why are the imports below the check for main

File: drivers/sub8_videoray_m5_thruster/nodes/thruster_spin_test.py
Comment (mattlangford) [inline drivers/sub8_videoray_m5_thruster/nodes/thruster_spin_test.py]: Does this actually sleep? I thought you had to call `sleep` or something

File: drivers/sub8_videoray_m5_thruster/nodes/thruster_spin_test.py
Comment (mattlangford) [reply drivers/sub8_videoray_m5_thruster/nodes/thruster_spin_test.py]: And everything for that matter.

This could be something that might be easier as an rqt plugin if Anthony is bent on making on

File: drivers/sub8_videoray_m5_thruster/nodes/thruster_spin_test.py
Comment (mattlangford) [inline drivers/sub8_videoray_m5_thruster/nodes/thruster_spin_test.py]: Maybe you could constrain the thrust here (`np.clip(thrust, 0.0, 1.0)`) instead of doing it each time you update.

File: drivers/sub8_videoray_m5_thruster/nodes/thruster_spin_test.py
Comment (mattlangford) [inline drivers/sub8_videoray_m5_thruster/nodes/thruster_spin_test.py]: Would it be better to trigger them by name rather than ID

File: drivers/sub8_videoray_m5_thruster/sub8_thruster_comm/thruster_comm.py
Comment (mattlangford) [inline drivers/sub8_videoray_m5_thruster/sub8_thruster_comm/thruster_comm.py]: Does thruster info exist anymore?

File: simulation/sub8_gazebo/src/sub8_thruster_config.cpp
Comment (mattlangford) [inline simulation/sub8_gazebo/src/sub8_thruster_config.cpp]: Redefinition of `i`

**Review by mattlangford — COMMENTED** [review]
Video Ray does make a distinction between `motor_id` and`node_id`. I don't remember which we actually used to address each thruster, but if it's `node_id`, probably best to leave it like it was.
Approval pending comment responses.

File: drivers/sub8_videoray_m5_thruster/nodes/thruster_spin_test.py
Comment (DSsoto) [reply drivers/sub8_videoray_m5_thruster/nodes/thruster_spin_test.py]: Fixed/.

It could be, but it's pretty nice and easy to use as it is. Not a good use of his time.

File: drivers/sub8_videoray_m5_thruster/nodes/thruster_spin_test.py
Comment (DSsoto) [reply drivers/sub8_videoray_m5_thruster/nodes/thruster_spin_test.py]: No because it's only one key press for each thruster so you can easily toggle each thruster. Plus they're all labeled anyways.

File: drivers/sub8_videoray_m5_thruster/nodes/thruster_spin_test.py
Comment (DSsoto) [reply drivers/sub8_videoray_m5_thruster/nodes/thruster_spin_test.py]: I did it down there so that I could display the new thrust value whenever it is changed.

File: drivers/sub8_videoray_m5_thruster/nodes/thruster_spin_test.py
Comment (DSsoto) [reply drivers/sub8_videoray_m5_thruster/nodes/thruster_spin_test.py]: Fixed

File: simulation/sub8_gazebo/src/sub8_thruster_config.cpp
Comment (DSsoto) [reply simulation/sub8_gazebo/src/sub8_thruster_config.cpp]: Good catch, fixed.

**Review by mattlangford — APPROVED** [review]
Did you roll with this last test day and see that everything is good?

Also again with the naming, Video Ray does make the distinction between node ID and motor ID - can you get with @duggerd and confirm we actually address them by motor ID?

Also I don't know if I should wait for CI to merge or something so I'll standby.

**Comment by DSsoto** [comment]
We ran this version of the repo at the pool and everything worked before thruster slip became an issue. I didn't manage to get the spin test utility running by test day, but I got it working last night so I can confirm that I can spin all of the thrusters by motor_id.

I just fixed the broken rostest, so this is ready to merge.

**Pull request merged**
